### DISCUSSION TITLE
Poll: Extreme Speed (LCM) vs Lightning

### DISCUSSION THREAD

**Discussion by mashb1t** [lllyasviel/Fooocus#2416] [Polls]
Quick check for https://github.com/lllyasviel/Fooocus/pull/2415 / https://github.com/lllyasviel/Fooocus/discussions/2318 comparing Extreme Speed (LCM) 8 steps vs Lightning 4 steps 
Images of each round were created using the same model / config / seed / styles / etc., only difference is the performance setting.

I'll post which one is which in a few days.

<details><summary>Round 1</summary>
<p>

prompt: otter mage wearing a wizard hat

A
![image](https://github.com/lllyasviel/Fooocus/assets/9307310/ceb442d6-4db2-40db-9073-50e958df4f68)

B
![image](https://github.com/lllyasviel/Fooocus/assets/9307310/843f0faa-08d4-4d87-b64d-ec2f161f600d)

</p>
</details> 

<details><summary>Round 2</summary>
<p>

prompt: portrait of a reahead woman, freckles, highly detailed

A
![image](https://github.com/lllyasviel/Fooocus/assets/9307310/baad6a67-a212-41d3-95a7-74d891a91c63)

B
![image](https://github.com/lllyasviel/Fooocus/assets/9307310/cbbf8361-6d48-4d1b-bc71-ca7c9a0f8e20)

</p>
</details> 

<details><summary>Round 3</summary>
<p>

prompt: purple flower, water drops, bee, pollen, golden hour

A
![image](https://github.com/lllyasviel/Fooocus/assets/9307310/550d62f8-4b2a-4b19-a554-a935fda04e48)

B
![image](https://github.com/lllyasviel/Fooocus/assets/9307310/7ecb9b89-8e93-4808-81ad-ccdf80142dd2)

</p>
</details> 

<details><summary>Round 4</summary>
<p>
prompt: beautiful landscape, mountains, lake, reflection, sunrise

A
![image](https://github.com/lllyasviel/Fooocus/assets/9307310/9dab03b2-d78d-471b-a37c-f313634f39b8)

B
![image](https://github.com/lllyasviel/Fooocus/assets/9307310/1a7bb06f-6f0d-47ac-85b9-690200187852)

</p>
</details>

**Comment by jaimeastorga2000** [comment]
Voted. Two comments:

You should post the prompts, so that we can judge prompt adherence as well as image quality.

Also, why are you comparing LCM 8 steps vs Lightning 4 steps? Lightning has an 8 step LoRA.

**Reply by mashb1t** [reply]
@jaimeastorga2000 Thank you for the feedback.
The goal of this poll is to see if lightning with improved performance (4 steps) can keep up with or outshine lcm (8 steps), so this can be added as fourth performance mode.

**Reply by wisdombox44** [reply]
that would be so nice Lightning image quality seems to better than Turbo, quality is okay, speed is amazing I am aslo playing with
Round 1 A
Round 2 B Looks to rough but stiil more natural than A
Round 3 A
Round 4 A

**Reply by IamTirion** [reply]
This is unrelated but, if 8-step Lightning is better than LCM, will you replace LCM with 8-step Lightning for the Extreme Speed option?

**Reply by mashb1t** [reply]
This decision has to be done carefully and backwards compatibility, especially now with metadata, has to be considered.
Most likely extreme speed will stay with LCM, but not 100% certain. There are no plans for now to switch the extreme speed preset.

**Comment by mashb1t** [comment]
Folks, here is the resolution:

- A is Lightning 4 steps LoRA
- B is Extreme Speed (LCM) - default 8 steps

It's crazy that Lightning manages to produce better results in comparison, even with half the steps. Still has some quirks when generating faces though.
Check out the PR for the new performance setting here: https://github.com/lllyasviel/Fooocus/pull/2415

**Comment by xhoxye** [comment]
New technologies are changing with each passing day
TCD Trajectory Consistency Distillation
https://github.com/jabir-zheng/TCD
TCD-SDXL-LoRA
https://huggingface.co/h1t/TCD-SDXL-LoRA/blob/main/pytorch_lora_weights.safetensors
cfg=1, euler_a, steps=8

**Reply by mashb1t** [reply]
@xhoxye TCD needs a separate sampler and scheduler to produce the best results, also see https://github.com/comfyanonymous/ComfyUI/issues/2985 and https://github.com/jabir-zheng/TCD/issues/7, while Lightning is directly compatible with the current pipeline implementation.

**Reply by xhoxye** [reply]
TCD the results produced without using a specific sampler and scheduler can also be better than lightning
![2024-03-10_14-00-58_4509](https://github.com/lllyasviel/Fooocus/assets/129571231/d88acfa9-6718-43fd-8b37-84da61f0ecc6)
![2024-03-10_14-11-03_6816](https://github.com/lllyasviel/Fooocus/assets/129571231/e2f155c3-65ce-420f-9a59-7d6d941606b8)

**Reply by mashb1t** [reply]
@xhoxye Can you please clarify what you mean by this?
> without using a specific sampler and scheduler

Sadly Github strips any metadata from the image on upload. Which parameters / samplers / etc. did you use for the images?

**Reply by xhoxye** [reply]
`{
"prompt": "otter mage wearing a wizard hat",
"negative_prompt": "",
"prompt_expansion": "otter mage wearing a wizard hat, fine detailed, elegant, intricate, light glowing, sharp focus, cinematic, highly saturated colors, radiant, enhanced quality, dynamic composition, epic gorgeous, stunning, creative, pure, attractive, scenic, rich deep background, intense, beautiful, amazing, spiritual, thought, inspiring, full color, illuminated, perfect, lovely, pretty",
"styles": "['Fooocus V2', 'Fooocus Enhance', 'Fooocus Sharp']",
"performance": "Speed",
"steps": 8,
"resolution": "(896, 1152)",
"guidance_scale": 1,
"sharpness": 0,
"adm_guidance": "(1, 1, 0)",
"base_model": "juggernautXL_v9Rundiffusionphoto2.safetensors",
"refiner_model": "None",
"refiner_switch": 0.5,
"sampler": "euler_ancestral",
"scheduler": "karras",
"seed": 5182389668215649573,
"lora_combined_1": "LCM\\TCD-SDXL-LoRA.safetensors : 1.0",
"metadata_scheme": "fooocus",
"version": "Fooocus v2.2.1 SimpleSDXL_v20240309.157a88a"
}`
![2024-03-10_16-22-55_8955](https://github.com/lllyasviel/Fooocus/assets/129571231/ee5b1dba-1724-4e03-b7ff-869732f123d9)
![QQ截图20240310162944](https://github.com/lllyasviel/Fooocus/assets/129571231/ccfc75b0-3051-401c-adff-3027be0fc0b1)

**Reply by mashb1t** [reply]
@xhoxye i did a few (lot) more testing with TCD and various samplers (sgm_uniform, euler a and a few more) as well as schedulers (karras works best for me) and a general result of mine is that TCD is following the prompt less than Lightning with same styles etc.

<details><summary>Examples</summary>
<p>

otter-wizard-lightning-sgm_uniform
![otter-wizard-lightning-sgm_uniform](https://github.com/lllyasviel/Fooocus/assets/9307310/c2db415a-2764-4512-a145-27f9d693781f)

otter-wizard-tcd-sgm_uniform
![otter-wizard-tcd-sgm_uniform](https://github.com/lllyasviel/Fooocus/assets/9307310/3a34a9d9-3132-45f6-99f1-d7c0174c253d)

otter-wizard-tcd-karras
![otter-wizard-tcd-karras](https://github.com/lllyasviel/Fooocus/assets/9307310/26b4aa67-daac-4d7e-9b40-f1f3f5ab8043)

landscape-lightning
![image - 2024-03-10T135458 875](https://github.com/lllyasviel/Fooocus/assets/9307310/2ac024c3-aad8-4e7d-ae59-885f9077816c)

landscape-tcd-karras
![image - 2024-03-10T135344 353](https://github.com/lllyasviel/Fooocus/assets/9307310/36f152bc-7ab2-416b-9acb-c959071a6d16)

</p>
</details> 

=> using lightning 4step LoRA in performance for now. TCD is more versatile (like LCM), as it comes as single LoRA, not like Lightning one LoRA for a specific amount of steps. Let's see how TCD is doing with the TCD sampler.

**Reply by mashb1t** [reply]
@xhoxye the tcd sampler has been implemented in https://github.com/lllyasviel/Fooocus/pull/2907, please check if the virtual scheduler makes sense or if we should rather use sgm_uniform.

**Comment by xhoxye** [comment]
`{
"prompt": "beautiful landscape, mountains, lake, reflection, sunrise",
"negative_prompt": "",
"prompt_expansion": "beautiful landscape, mountains, lake, reflection, sunrise, vivid colors, intricate, elegant, highly detailed, very sharp focus, professional, cinematic, surreal, still, beautiful, futuristic, open background, magic, thought, elite, rich, dramatic, atmosphere, romantic, enhanced, iconic, fine, epic, best, unique, ambient, dynamic, artistic, positive",
"styles": "['Fooocus V2', 'Fooocus Enhance', 'Fooocus Sharp']",
"performance": "Speed",
"steps": 8,
"resolution": "(1024, 1024)",
"guidance_scale": 1,
"sharpness": 0,
"adm_guidance": "(1, 1, 0)",
"base_model": "juggernautXL_v9Rundiffusionphoto2.safetensors",
"refiner_model": "None",
"refiner_switch": 0.5,
"sampler": "euler_ancestral",
"scheduler": "karras",
"seed": 6256444360585980502,
"lora_combined_1": "LCM\\TCD-SDXL-LoRA.safetensors : 1.0",
"metadata_scheme": "fooocus",
"version": "Fooocus v2.2.1 SimpleSDXL_v20240309.157a88a"
}`
![2024-03-10_16-26-17_5588](https://github.com/lllyasviel/Fooocus/assets/129571231/09327168-9e9d-4b8c-b8c7-70a160c57bfb)

**Comment by wangfeng35** [comment]
TCD is awesome, hope it will be incorporated into Fooocus!!!

**Comment by lgwjames** [comment]
For me

b
b
a
a

**Comment by Biglo2001** [comment]
My preferences are A, B, A, A
